lib/util/regions.js

```diff
@@ -416,38 +425,56 @@ export function getRegionsFromOverlaps(overlaps, deviceInfo) {
 
         regions = regions.includes(region) ? regions : regions.push(region);
     });
+    regions = regions.filter(r => isRegionInCore(r, coreInfo));
 
     return regions;
-}
+};
 
 /**
  * Given an instance of MemoryMap, content of loaded files and DeviceDefinition,
  * return the heuristically detected regions for loaded files.
  *
  * @param {Array}              memMaps      the memory map
- * @param {DeviceDefinition}   deviceInfo   the device infomation
+ * @param {DeviceDefinition}   coreInfo   the device infomation
  *
  * @returns {List} the list of region
  */
-export function getFileRegions(memMaps, deviceInfo) {
+export const getFileRegions = (memMaps, coreInfo) => {
     const overlaps = MemoryMap.overlapMemoryMaps(memMaps);
-    const regions = getRegionsFromOverlaps(overlaps, deviceInfo);
+    const regions = getRegionsFromOverlaps(overlaps, coreInfo);
 
     return regions;
-}
+};
 
 /**
  * Given an instance of MemoryMap, content of loaded device and DeviceDefinition,
  * return the heuristically detected regions for loaded device.
  *
  * @param {Array}              memMaps      the memory map
- * @param {DeviceDefinition}   deviceInfo   the device infomation
+ * @param {DeviceDefinition}   coreInfo   the device infomation
  *
  * @returns {List} the list of region
  */
-export function getTargetRegions(memMaps, deviceInfo) {
+export const getCoreRegions = (memMaps, coreInfo) => {
     const overlaps = MemoryMap.overlapMemoryMaps(memMaps);
-    const regions = getRegionsFromOverlaps(overlaps, deviceInfo);
+    const regions = getRegionsFromOverlaps(overlaps, coreInfo);
 
     return regions;
-}
+};
+
+/**
+ * Given an instance of MemoryMap, content of loaded device and DeviceDefinition,
+ * return the heuristically detected regions for loaded device.
+ *
+ * @param {Array}              memMaps      the memory map
+ * @param {DeviceDefinition}   deviceInfo   the device infomation
+ *
+ * @returns {List} the list of region
+ */
+export const getTargetRegions = (memMaps, deviceInfo) => {
+    let regions = List();
+    deviceInfo.cores.forEach(c => {
+        regions = regions.concat(getCoreRegions(memMaps, c));
+    });
+    return regions;
+};
```

Comment:
I think this would be the same:
```js
export const getTargetRegions = (memMaps, { cores }) => (
    new List(cores.map(c => getCoreRegions(memMaps, c)));
);
```

Reply:
👍